src/extractor/guidance/turn_handler.cpp

```diff
@@ -31,6 +51,67 @@ bool isEndOfRoad(const ConnectedRoad &,
            angularDeviation(possible_right_turn.angle, possible_left_turn.angle) >
                2 * NARROW_TURN_ANGLE;
 }
+
+template <typename Iterator>
+Iterator findOutermostForkCandidate(const Iterator start, const Iterator end)
+{
+    const auto outermost = std::adjacent_find(start, end, isOutermostForkCandidate);
+    if (outermost != end)
+    {
+        return outermost;
+    }
+    // if all roads are part of a fork, set `candidate` to the last road
+    else
+    {
+        return outermost - 1;
+    }
+}
+
+struct StraightestTurnAtIntersection
+{
+    StraightestTurnAtIntersection(std::size_t id, double deviation_from_straight)
+        : id(id), deviation_from_straight(deviation_from_straight){};
+
+    std::size_t id;
+    double deviation_from_straight;
+};
+
+// returns a tuple {i, angle_dev} of the road at the intersection that is closest to going straight:
+//     `intersection[i]` is the actual road
+//     `angle_dev` is the angle between going straight and taking the road `intersection[i]`
+StraightestTurnAtIntersection findClosestToStraight(const Intersection &intersection)
```

Comment:
can be removed completely and replaced with `findClosestTurn(STRAIGHT_ANGLE)` in `intersection`